### PR TITLE
UI fixes for jira ML-260

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
Issues fixed are # 24,25 & 27

### CODE REVIEW

File: apps/ml/site/home/home.jag [hunk 1]
```diff
@@ -182,9 +190,33 @@ $( document ).ready(function() {
...
+                }
+                else{
+                    count = data.length;
+                }
+            },
+            error: function(res) {
+                var errorText = JSON.parse(res.responseText)["exception"];
+                handleNotification(errorText, '#notification-area', 'warning');
+            }
+
```
Comment (cdathuraliya) [inline apps/ml/site/home/home.jag:211]: Please remove extra lines.

File: apps/ml/site/project/create_project.jag [hunk 2]
```diff
@@ -301,6 +307,14 @@ $( document ).ready(function() {
         return false;       
 	}
 
+	function isValidDataset(text){
+		if(text && text.length > 0 && text != "Select.."){
+			return true;
+		}
+		$('#dataset-selection-error').text('* Dataset is required.');
+        return false;
```
Comment (cdathuraliya) [inline apps/ml/site/project/create_project.jag:315]: Please format code.

File: apps/ml/site/project/projects.jag [hunk 3]
```diff
@@ -441,23 +444,30 @@ include("../../includes/tenantAware.jag");
...
+                                '<div style="display:inline-flex"  data-toggle="tooltip" title="'+title+'">'+
+                                    '<p><a href="create_project.jag" class="cu-level2-btn btn-add-new-dark ' + appendCreateprojectDisabled(noOfDatasets) + '" data-toggle="tooltip" data-original-title="Create Project"><span>Create Project</span></a></p>'+
+                                '</div>'+
                             '</div>';
+
         		    } else {
         			    accordionPanelNoProjects = '<div class="ctrl-info-panel col-md-12 col-centered">'+
                                 '<h2>You do not have any projects created for dataset [' + datasetName + '] at the moment.</h2>'+
-                                '<p><a href="create_project.jag?datasetName=' + datasetName + '" class="cu-level2-btn btn-add-new-dark" data-toggle="tooltip" data-original-title="Create Project"><span>Create Project</span></a></p>'+
+                                '<p><a href="create_project.jag?datasetName=' + datasetName + '" class="cu-level2-btn btn-add-new-dark" data-toggle="tooltip" title="hello" data-original-title="Create Project"><span>Create Project</span></a></p>'+
```
Comment (cdathuraliya) [inline apps/ml/site/project/projects.jag:470]: Add a meaningful title attribute value or remove if it unused.

Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:470]: fixed with PR https://github.com/wso2/carbon-ml/pull/200

File: apps/ml/site/home/home.jag (hunk 1 above)
Comment (fazlan-nazeem) [inline apps/ml/site/home/home.jag:211]: fixed with PR https://github.com/wso2/carbon-ml/pull/200

File: apps/ml/site/project/create_project.jag (hunk 2 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/create_project.jag:315]: was fixed by a later commit

File: apps/ml/site/project/projects.jag [hunk 4]
```diff
@@ -441,23 +444,30 @@ include("../../includes/tenantAware.jag");
     }
 
     function loadProjectsOfDataset(datasetName) {
+        var noOfDatasets = getAvailableDatasetCount();
+        var title="";
+        if(noOfDatasets == 0){
```
Comment (lafernando) [inline apps/ml/site/project/projects.jag:449]: Fix formatting.

File: apps/ml/site/project/projects.jag [hunk 5]
```diff
@@ -189,6 +189,9 @@ include("../../includes/tenantAware.jag");
 
         // load projects and analyses to accordion
         loadProjects();
+
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:192]: Remove unnecessary newlines.

File: apps/ml/site/project/projects.jag [hunk 6]
```diff
@@ -595,12 +609,30 @@ include("../../includes/tenantAware.jag");
...
     // function to append button disabled CSS class when there are no analyses in a project
     function appendDisabled(noOfAnalyses) {
+
         var classAppend = '';
         if(noOfAnalyses == 0) {
             classAppend = 'btn-disabled';  
         }
         return classAppend;
     }
+    // function to append button disabled CSS class when there are no complete models in a project
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:619]: Use proper linguistics for method comments. Use capitals to begin the sentences.

File: apps/ml/site/project/projects.jag [hunk 7]
```diff
@@ -595,12 +609,30 @@ include("../../includes/tenantAware.jag");
...
+    function appendCompareModelsDisabled(noOfCompleteModels) {
+        var classAppend = '';
+        if(noOfCompleteModels == 0) {
+            classAppend = 'btn-disabled';  
+        }
+        return classAppend;
+    }
+
+   // function to append button disabled CSS class when there are no datasets to create a project
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:628]: Same goes here.

File: apps/ml/site/project/projects.jag [hunk 8]
```diff
@@ -595,12 +609,30 @@ include("../../includes/tenantAware.jag");
...
+   function appendCreateprojectDisabled(noOfDatasets) {
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:629]: Formatting errors. Please fix.

File: apps/ml/site/project/projects.jag [hunk 9]
```diff
@@ -595,12 +609,30 @@ include("../../includes/tenantAware.jag");
...
+        var classAppend = '';
+        if(noOfDatasets == 0) {
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:631]: More formatting issues.

File: apps/ml/site/project/projects.jag [hunk 10]
```diff
@@ -663,6 +695,51 @@ include("../../includes/tenantAware.jag");
...
+                }
+                else{
+                    count = data.length;
+                } 
+            },
+            error: function(res) {
+                var errorText = JSON.parse(res.responseText)["exception"];
+                handleNotification(errorText, '#notification-area', 'warning');
+            }
+
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:716]: Remove unnecessary newlines.

File: apps/ml/site/project/projects.jag [hunk 11]
```diff
@@ -663,6 +695,51 @@ include("../../includes/tenantAware.jag");
...
+
+        });
+        return count;
+    }   
+
+    function getCompleteModelsCount(projectId){
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:722]: More formatting issues.

File: apps/ml/site/project/projects.jag [hunk 12]
```diff
@@ -663,6 +695,51 @@ include("../../includes/tenantAware.jag");
...
+         var count;
+         $.ajax({
+            type: "GET",
+            url: serverUrl+"/api/projects/"+projectId+"/models",
+            async: false,
+            success: function(data) {
+              if(data.length == 0){
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:729]: Some more.

File: apps/ml/site/project/projects.jag [hunk 13]
```diff
@@ -663,6 +695,51 @@ include("../../includes/tenantAware.jag");
...
+                count = 0;
+              }
+              else{
```
Comment (prabathabey) [inline apps/ml/site/project/projects.jag:732]: A few more

File: apps/ml/site/project/projects.jag (hunk 5 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:192]: Removed unnecessary lines with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 6 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:619]: Fixed with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 7 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:628]: fixed with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 8 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:629]: fixed with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 9 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:631]: fixed with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 10 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:716]: Fixed with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 11 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:722]: Fixed with PR https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 12 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:729]: Fixed with https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag (hunk 13 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:732]: Fixed with https://github.com/wso2/carbon-ml/pull/201

File: apps/ml/site/project/projects.jag [hunk 14]
```diff
@@ -595,12 +609,30 @@ include("../../includes/tenantAware.jag");
 
     // function to append button disabled CSS class when there are no analyses in a project
     function appendDisabled(noOfAnalyses) {
+
```
Comment (sinthuja) [inline apps/ml/site/project/projects.jag:612]: Remove unnecessary new lines.

File: apps/ml/site/home/home.jag [hunk 15]
```diff
@@ -152,6 +154,12 @@ $( document ).ready(function() {
         window.location.href = requestedPage;
     }
 
+    var datasetCount = getAvailableDatasetCount();
+    if (datasetCount == 0){
```
Comment (Kishanthan) [inline apps/ml/site/home/home.jag:158]: Formatting issue here.

File: apps/ml/site/project/projects.jag [hunk 16]
```diff
@@ -595,12 +609,30 @@ include("../../includes/tenantAware.jag");
...
         var classAppend = '';
         if(noOfAnalyses == 0) {
             classAppend = 'btn-disabled';  
         }
         return classAppend;
     }
+    // function to append button disabled CSS class when there are no complete models in a project
+    function appendCompareModelsDisabled(noOfCompleteModels) {
+        var classAppend = '';
+        if(noOfCompleteModels == 0) {
```
Comment (Kishanthan) [inline apps/ml/site/project/projects.jag:622]: Formatting issue here.

File: apps/ml/site/project/projects.jag [hunk 17]
```diff
@@ -663,6 +695,51 @@ include("../../includes/tenantAware.jag");
...
+        var count;
+        $.ajax({
+            type: "GET",
+            url: serverUrl+"/api/datasets?status=Available",
+            async: false,
+            success: function(data) {
+                if(data.length == 0){              
+                    count =0;
+                }
+                else{
```
Comment (Kishanthan) [inline apps/ml/site/project/projects.jag:708]: Formatting issue. "else" statement should start right after the "if" closing curly brackets.

File: apps/ml/site/project/projects.jag [hunk 18]
```diff
@@ -663,6 +695,51 @@ include("../../includes/tenantAware.jag");
...
     }
 
+    function getAvailableDatasetCount(){
+        var count;
+        $.ajax({
+            type: "GET",
+            url: serverUrl+"/api/datasets?status=Available",
+            async: false,
+            success: function(data) {
+                if(data.length == 0){
```
Comment (Kishanthan) [inline apps/ml/site/project/projects.jag:705]: Formatting issue.

Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:705]: fixed with PR  https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/project/projects.jag (hunk 17 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:708]: fixed with https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/project/projects.jag (hunk 16 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:622]: fixed with https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/home/home.jag (hunk 15 above)
Comment (fazlan-nazeem) [inline apps/ml/site/home/home.jag:158]: fixed with https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/project/projects.jag (hunk 14 above)
Comment (fazlan-nazeem) [inline apps/ml/site/project/projects.jag:612]: fixed with https://github.com/wso2/carbon-ml/pull/202

**Pull request merged**